src/FurCoNZ.Web/Services/ReminderService.cs

```diff
@@ -0,0 +1,153 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace FurCoNZ.Web.Services
+{
+    public class ReminderService : IReminderService
+    {
+        private readonly FurCoNZDbContext _dbContext;
+        private readonly IEmailService _emailService;
+        private readonly ILogger<ReminderService> _logger;
+        private readonly ReminderServiceOptions _options;
+        private readonly IViewRenderService _viewRenderService;
+
+        public ReminderService(
+            FurCoNZDbContext dbContext,
+            IEmailService emailService,
+            ILogger<ReminderService> logger,
+            IOptions<ReminderServiceOptions> options,
+            IViewRenderService viewRenderService
+        )
+        {
+            _dbContext = dbContext;
+            _emailService = emailService;
+            _logger = logger;
+            _options = options.Value;
+            _viewRenderService = viewRenderService;
+        }
+
+        public async Task Send30DayRemainingPendingOrdersAsync(CancellationToken cancellationToken = default)
+        {
+            // Get the last time the reminders were sent
+            var lastSend = await _dbContext.RemindersLastRuns.FindAsync(new[] { "SendUnpaidReminderForOrders" }, cancellationToken);
+            if (lastSend == null)
+            {
+                lastSend = new Models.RemindersLastRun
+                {
+                    ReminderService = "SendUnpaidReminderForOrders",
+                    LastRun = DateTimeOffset.MinValue,
+                };
+                _dbContext.RemindersLastRuns.Add(lastSend);
+            }
+            var lastRun = lastSend.LastRun;
+            var currentRun = DateTimeOffset.Now;
+
+            // Get the orders that we will send reminders for
+            // TODO: Support multiple reminders before expiry
+            var expiredOrdersSinceLastSend = await _dbContext.Orders.AsNoTracking()
+                .Where(o =>
+                    !o.Audits.Any() && // No payments have been made
+                    o.CreatedAt.AddDays(_options.UnpaidOrderReminderDays.First()) <= currentRun && // Order has expired
+                    o.CreatedAt.AddDays(_options.UnpaidOrderReminderDays.First()) >= lastRun // Reminder has not been sent
+                )
+                .Select(o => new
+                {
+                    o.Id,
+                    o.CreatedAt,
+                    o.OrderedBy.Name,
+                    o.OrderedBy.Email
+                })
+                .ToListAsync(cancellationToken);
+
+            // Send emails for each order
+            foreach (var order in expiredOrdersSinceLastSend)
+            {
+                // Gather metadata
+                var toAddresses = new MailAddressCollection
+                {
+                    new MailAddress (order.Email, order.Name),
+                };
+                var subject = $"Order #{order.Id} has expired";
```

Comment:
Is it possible to set the subject from the rendered view's `ViewData["Title"]` to add flexibility to the templates and avoid a hard coded Subject

Reply:
Great name :+1: